python/ray/actor.py

```diff
@@ -762,31 +756,6 @@ def _remote(self, args=None, kwargs=None, **actor_options):
                 "'non_detached' and 'None'."
             )
 
-        # Set the actor's default resources if not already set. First three
-        # conditions are to check that no resources were specified in the
-        # decorator. Last three conditions are to check that no resources were
-        # specified when _remote() was called.
-        if (
-            num_cpus is None
```

Comment:
The original implementation is kind of buggy: It does not consider the case of "memory". "memory" should also be considered as a resource.